geolocator_android/android/src/main/java/com/baseflow/geolocator/location/BackgroundNotification.java

```diff
@@ -0,0 +1,102 @@
+package com.baseflow.geolocator.location;
+
+import android.annotation.SuppressLint;
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+public class BackgroundNotification {
+    private Context context;
+    private Integer notificationId;
+    private String channelId;
+    private ForegroundNotificationOptions options;
+    private NotificationCompat.Builder builder;
+
+    public BackgroundNotification(
+            Context context,
+            String channelId ,
+            Integer notificationId,
+            ForegroundNotificationOptions options
+    ) {
+        this.context = context;
+        this.notificationId = notificationId;
+        this.channelId = channelId;
+        builder = new NotificationCompat.Builder(context, channelId)
+                .setPriority(NotificationCompat.PRIORITY_HIGH);
+        updateNotification(options, false);
+    }
+
+    private int getDrawableId(String iconName, String defType) {
+        return context.getResources().getIdentifier(iconName, defType, context.getPackageName());
+    }
+
+    @SuppressLint("UnspecifiedImmutableFlag")
+    private PendingIntent buildBringToFrontIntent() {
+        Intent intent = context.getPackageManager()
+                .getLaunchIntentForPackage(context.getPackageName());
+
+        if (intent != null) {
+            intent.setPackage(null);
+            intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_RESET_TASK_IF_NEEDED);
+            int flags = PendingIntent.FLAG_UPDATE_CURRENT;
+            if (Build.VERSION.SDK_INT > 23) {
```

Comment:
Use `Build.VERSION_CODES.M` instead to the `23` number literal:

```suggestion
            if (Build.VERSION.SDK_INT > Build.VERSION_CODES.M) {
```